packages/now-client/src/utils/index.ts

```diff
@@ -75,6 +76,55 @@ const maybeRead = async function<T>(path: string, default_: T) {
   }
 };
 
+export async function readdirRelative(
+  path: string,
+  ignores: string[]
+): Promise<string[]> {
+  const preprocessedIgnores = ignores.map(ignore => {
+    if (ignore.endsWith('/')) {
+      return ignore.slice(0, -1);
+    }
+    return ignore;
+  });
+  const dirContents = await readdir(path, preprocessedIgnores);
+  return dirContents.map(filePath => relative(process.cwd(), filePath));
+}
+
+export async function buildFileTree(
+  path: string | string[],
+  isDirectory: boolean,
+  debug: Debug
+): Promise<string[]> {
+  // Get .nowignore
+  let { ig, ignores } = await getVercelIgnore(path);
+
+  debug(`Found ${ig.ignores.length} rules in .nowignore`);
+
+  let fileList: string[];
+
+  debug('Building file tree...');
+
+  if (isDirectory && !Array.isArray(path)) {
+    // Directory path
+    const relativeFileList = await readdirRelative(path, ignores);
+    fileList = ig
+      .filter(relativeFileList)
+      .map((relativePath: string) => join(process.cwd(), relativePath));
```

Comment:
Let's move `const cwd = process.cwd()` to the top so its not called for every file. Similarly, it can be passed as an argument to `readdirRelative()`.